internal/scaffold/role.go

```diff
@@ -143,6 +145,72 @@ func UpdateRoleForResource(r *Resource, absProjectPath string) error {
 	}
 }
 
+// MergeHelmRoleForResource ...
+func MergeHelmRoleForResource(r *Resource, absProjectPath string, roleScaffold Role) error {
+	// append rbac rule to deploy/role.yaml
+	roleFilePath := filepath.Join(absProjectPath, DeployDir, RoleYamlFile)
+	roleYAML, err := ioutil.ReadFile(roleFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read role manifest %v: %v", roleFilePath, err)
+	}
+	obj, _, err := cgoscheme.Codecs.UniversalDeserializer().Decode(roleYAML, nil, nil)
+	if err != nil {
+		return fmt.Errorf("failed to decode role manifest %v: %v", roleFilePath, err)
+	}
+	switch role := obj.(type) {
+	// TODO: use rbac/v1..
+	case *rbacv1.Role:
+		//pri := &rbacv1.PolicyRule{}
+		//prj := &rbacv1.PolicyRule{}
+
+		for j := range roleScaffold.CustomRules {
+			prj := &roleScaffold.CustomRules[j]
+		iLoop:
+			for i := range role.Rules {
+				pri := &role.Rules[i]
+				// check if apiGroup and Verbs exists for new resource.
+				apiGroupFound := ifMatches(pri.APIGroups, prj.APIGroups)
+				verbFound := ifMatches(pri.Verbs, prj.Verbs)
+				resourceNamesFound := ifMatches(pri.ResourceNames, prj.ResourceNames)
+				if apiGroupFound && verbFound && resourceNamesFound {
+					for _, newResource := range prj.Resources {
+						resourceFound := findResource(pri.Resources, newResource)
+						if !resourceFound {
+							pri.Resources = append(pri.Resources, newResource)
+						}
+					}
+					break iLoop
+				}
+			}
+		}
+		return updateRoleFile(&role, roleFilePath)
+
+	default:
+		return errors.New("failed to parse role.yaml as a role")
+	}
+}
+
+func ifMatches(pr []string, newPr []string) bool {
+	if len(pr) != len(newPr) {
+		return false
+	}
+	for i, v := range pr {
+		if v != newPr[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func findResource(resourceSlice []string, r string) bool {
+	for _, newResource := range resourceSlice {
```

Comment:
Nit: `newResource` is a bit misleadingly named since we're passing `pri.Resources` which are our existing resources.

How about:

```go
findResource(resources []string, search string) bool {
    for _, r := range resources { ... }
    ...
```